nixos/modules/services/monitoring/prometheus/default.nix

```diff
@@ -4,8 +4,11 @@ with lib;
 
 let
   cfg = config.services.prometheus;
+  cfg2 = config.services.prometheus2;
   promUser = "prometheus";
   promGroup = "prometheus";
+  prom2User = "prometheus2";
```

Comment:
I suspect it would be acceptable (preferable, even?) to have prometheus and prometheus2 share the same UID/GID, unless there's a reason that won't work.

Reply:
@benley It's centainly preferable to have just one user. As of now the data dir /var/lib/prometheus(2) is also the home of the user and that's is used to create the dir and to ensure that the dir is automatically owned by it. I have yet to find out what's the best way to proceed if the data dirs are two. What you advise me to do?

Reply:
Hrm, good question.  I suppose you could put prometheus 2 data in a subdirectory of prometheus 1's data, but that introduces some danger when someone goes to delete their old prometheus 1 data dir once they're finished migrating.  Maybe the systemd unit could create the new data dir before starting the prometheus daemon?